### DISCUSSION TITLE
Chances to adding more features/options into the 3D viewer?

### DISCUSSION THREAD

**Discussion by ZoruaLoL** [4sval/FModel#453] [Q&A] [answered]
Hi, I've been using FModel for a while and something that I appreciate over UE Viewer is how much better models tend to be rendered material-wise

That being said, there are certain features from UE Viewer that I sorely miss when using FModel that make the experience to much inconvenient:
- There's no way to adjust the FOV
- There's no screenshot button (could you maybe even add a gif/video recorder for animations?)
- Background options are very lackluster, choosing between the skybox and pure black, UE Viewer's grey gradient is much preferable
- Moving around the viewport is pretty cumbersome with the keyboard, the mouse should be able to be used to zoom and move around
- Loading animations is cumbersome as it has to be done one-by-one instead of automatically appending all animations associated to a model

So please consider implementing any of these, I think it would close the gap between programs for people coming from UE Viewer

**Comment by 4sval** [comment, marked as answer]
Hello @ZoruaLoL,

- X/C can be used to adjust the FOV
- ShareX/OBS exist. If you're talking about transparent background screenshots, that's not the purpose of FModel, especially with the current generic material implementation. Use Blender.
- That's personal preference, adding more options would be cosmetic
- I'm surprised by this one, we offer 2 different camera options. In Arcball mode the camera is centered on the last loaded model and the mouse can be used to move around the model while keeping it perfectly centered. W/S can be used to move closer/further, Q/E to go up/down, A/D to move the center point left/right. In Fly Cam mode it works as in any video game and is auto selected while loading levels. I don't think that's cumbersome, the mouse is not the only input method a computer has, and umodel is far from an example in term of UI/UX.
- That's a strength umodel has, and this goes with the fact FModel also can't "scan" packages. It's just way too slow in C# even with everything optimized for this use case.

Don't expect screenshots and other camera options, other stuff might come sooner or later.

**Reply by ZoruaLoL** [reply]
Thanks for your response, but:

- X/C zooms into the image instead only adjusting the FOV while keeping the relation between the model and the camera in the viewport the same the same like umodel does, commonly known as a "Dolly zoom" effect
- - On a side note, could at least an option to set the projection to "ortographic" be possible?
- Yes I know about the Arcball mode, but I feel there are still some things that the mouse can do much better than the keys because **they're just too slow** (trying to zoom out after setting the FOV to the lowest possible is such a drag). And keep in mind the way umodel uses the mouse is closer to how 3D software such as Blender uses it (where "walk navigation" WASD controls aren't the default) than anything in FModel

**Reply by 4sval** [reply]
- Zooming in and out is more than enough. You're controlling a camera, not a player, and this is not a game, nor a renderer engine, not even near a game engine. It's the first version of a 3D Viewer made to preview models we didn't make with shaders we can't decompile. What would be the point of being able to adjust the FOV or have an orthographic camera?
- Ok the scroll wheel may be useful for zooming and panning. If you feel that's too slow you can press Shift or adjust the speed under the World tab, but again you're not supposed to always zoom in and out or pan 500m away from the model (in Arcball mode). FModel's camera acts like a noclip camera in games and this will not change in the next refactor.

What I keep in mind tho is:
- panning with MMB
- zooming with MWD/MWU
- having camera controls and other world controls overlayed at the top of the viewport
- being able to lock the Arcball camera to the model so that while animations play, you don't have to move
- being able to animate multiple models at the same time
- preview other LODs
- etc
